compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -559,20 +509,26 @@ fn link_staticlib<'a>(
                         return true;
                     }
 
-                    // Otherwise if this is *not* a rust object and we're skipping
-                    // objects then skip this file
-                    if skip_object_files
-                        && (!fname.starts_with(&obj_start) || !fname.ends_with(".o"))
-                    {
+                    // Skip objects for bundled libs.
+                    if bundled.iter().any(|l| l.as_str() == fname) {
                         return true;
                     }
 
-                    // ok, don't skip this
                     false
                 }),
             )
             .unwrap();
 
+            let bundled: FxHashSet<_> = native_libs.iter().filter_map(|lib| lib.filename).collect();
+            archive_builder_builder
+                .extract_bundled_libs(path, tempdir.as_ref(), &bundled)
+                .unwrap_or_else(|e| sess.emit_fatal(e));
+            for name in bundled {
+                let joined = tempdir.as_ref().join(name.as_str());
+                let path = joined.as_path();
+                ab.add_archive(path, Box::new(|_| false)).unwrap();
```

Comment:
Where is `cfg` (`relevant_lib`) applied to the bundled libs now?
From what I see they are always added regardless of `cfg`.